server/routes/private.js

```diff
@@ -49,6 +50,41 @@ router.post('/api/book-appointment', async (req, res) => {
 
 });
 
+router.post('/api/recurring_schedule', async (req, res) => {
+    // Validate appropriate parameters are passed into the book appointment endpoint.
+    // Verification to make sure appointments are only made for future dates and that the worker is available during that time will be handled by the front end.
+    const paramSchema = Joi.object({
+        
+        workerTimeslotId: Joi.number().integer().required(),
+        slotId: Joi.number().integer().required(),
+        schoolId: Joi.number().integer().required(),
+        workerId: Joi.number().integer().required(), 
+        status:Joi.string().max(300).required(),
+        date: Joi.string().allow(null).max(300)
+
+        
+    });
+
+    const query = req.query ? req.query : {};
+
+    const workerTimeslotId = query.workerTimeslotId ? query.workerTimeslotId : null;
+    const slotId = query.slotId ? query.slotId : null;
+    const schoolId = query.schoolID ? query.schoolID : null;
+    const workerId = query.workerID ? query.workerID : null;
+    const status = query.status ? query.status : null;
+    const date = query.date ? query.date: null; 
+
+    const { error } = paramSchema.validate({workerTimeslotId, slotId, schoolId, workerId, status, date});
+
+    if (!_.isNil(error)) res.send(error);
+
+    // Attempts to insert the appointment into the database.
```

Comment:
I would adjust the comment to say add worker availability instead of appointment to avoid confusion.

Reply:
Thank you @catai9 I've added the change!